python_frank_energie/models.py

```
@@ -3148,10 +3148,20 @@ def from_be_dict(cls, data: dict[str, Any]) -> 'MarketPrices':
+                "BE market prices: 'data' key is absent or null "
+                "(prices likely not yet published). Returning empty prices."
+            )
+            return cls(PriceData(), PriceData())
+
         try:
-            payload = data.get("data").get("marketPrices", {})
-        except KeyError as err:
-                    raise ValueError(f"Invalid response format: %s" % err) from err
+            payload = data_inner.get("marketPrices", {})
+        except (KeyError, AttributeError) as err:
+            raise ValueError(f"Invalid response format: %s" % err) from err
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)